src/main/java/com/hivemq/cli/commands/options/TlsOptions.java

```diff
@@ -138,104 +127,255 @@ private boolean useBuiltSslConfig() {
+                    clientKeystorePrivateKeyPassword);
+        } else if (onlyCertificateExists) {
+            throw new RuntimeException(
+                    "Only the client certificate parameter exists. Please add the private key parameter as well.");
+        } else if (onlyPrivateKeyExists) {
+            throw new RuntimeException(
+                    "Only the private key parameter exists. Please add the client certificate parameter as well.");
+        } else if (certificateAndPrivateKeyExists) {
+            final Collection<X509Certificate> x509Certificates =
+                    TlsUtil.getCertificateChainFromFile(clientCertificatePath);
+            final PrivateKey privateKey = TlsUtil.getPrivateKeyFromFile(clientPrivateKeyPath, clientPrivateKeyPassword);
+            final KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
+            ks.load(null, null);
+            final String createdKeystoreName = "privateKeyAlias";
+            final String createdKeystorePassword = "keystorePassword";
+            ks.setKeyEntry(createdKeystoreName,
+                    privateKey,
+                    createdKeystorePassword.toCharArray(),
+                    x509Certificates.toArray(new X509Certificate[0]));
+
+            keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            keyManagerFactory.init(ks, createdKeystorePassword.toCharArray());
+        } else if (defaultKeystoreExists) {
+            if (defaultCertificateAndPrivateKeyExists) {
+                Log.warn("The keystore default option (properties file) shadows the private key " +
+                        "corresponding certificate default options (properties file). " +
+                        "If you want to use them instead please remove the keystore default option.");
+            }
+            final String defaultKeystorePassword = defaultCLIProperties.getKeystorePassword();
+            final String defaultKeystorePrivateKeyPassword = defaultCLIProperties.getKeystorePrivateKeyPassword();
             try {
-                clientCertificateChain = defaultCLIProperties.getClientCertificateChain();
+                keyManagerFactory = TlsUtil.createKeyManagerFactoryFromKeystore(defaultKeystore,
+                        defaultKeystorePassword,
+                        defaultKeystorePrivateKeyPassword);
             } catch (final Exception e) {
                 Logger.error(e,
-                        "Default client certificate chain could not be loaded ({})",
+                        "Default keystore (properties file) could not be loaded ({})",
                         Throwables.getRootCause(e).getMessage());
+                throw e;
             }
-        }
-
-        if (clientPrivateKey == null) {
+        } else if (onlyDefaultCertificateExists) {
+            throw new RuntimeException("Only the client certificate default option (properties file) exists. " +
```

Comment:
Include the exact name of the default option and the location of the file please.